Makefile

```diff
@@ -816,11 +817,14 @@ release-docs: | $(DIST_DIRS) docs
 	tar -czf $(DIST)/release/gitea-docs-$(VERSION).tar.gz -C ./docs/public .
 
 .PHONY: docs
-docs:
+docs: deps-docs
+	cd docs; make trans-copy clean build-offline;
+
+.PHONY: deps-docs # TODO: find a way to install hugo without needing sudo permissions and add deps-docs to deps (the next block)
+deps-docs:
 	@hash hugo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
 		curl -sL https://github.com/gohugoio/hugo/releases/download/v$(HUGO_VERSION)/hugo_$(HUGO_VERSION)_Linux-64bit.tar.gz | tar zxf - -C /tmp && mv /tmp/hugo /usr/bin/hugo && chmod +x /usr/bin/hugo; \
```

Comment:
Do we really need to put `hugo` in `/usr/bin`? I wonder if there would be a better place to place this binary and call it?

For example couldn't it go in to `~/go/bin` and have `~/go/bin` be added on to the PATH?